src/logic/calculate.js

```diff
@@ -0,0 +1,15 @@
+import operate from './operate';
+
+const calculate = ({ total, next, operation }, buttonName) => {
+  let result = total;
+  let nextNumber = next;
+  if (buttonName === '+/-') {
+    nextNumber *= -1;
+    result += -1;
+    return { result, nextNumber };
+  }
+  result = operate(result, nextNumber, operation);
+  return { result, nextNumber };
+};
```

Comment:
- [ ] Here at the calculate function we need to cover all cases for the calculator buttons, including the digits, operators, `AC`, `=`, `.`, 
![Screenshot from 2021-04-06 15-32-00](https://user-images.githubusercontent.com/30977845/113761001-51e04580-96ed-11eb-9972-7a994ea05490.png)